lib/elixir/lib/kernel.ex

```diff
@@ -2381,13 +2381,14 @@ defmodule Kernel do
     raise ArgumentError, "expected 0 or 1 argument for @#{name}, got: #{length(args)}"
   end
 
-  defp typespec(:type),          do: :deftype
-  defp typespec(:typep),         do: :deftypep
-  defp typespec(:opaque),        do: :defopaque
-  defp typespec(:spec),          do: :defspec
-  defp typespec(:callback),      do: :defcallback
-  defp typespec(:macrocallback), do: :defmacrocallback
-  defp typespec(_),              do: false
+  defp typespec(:type),               do: :deftype
+  defp typespec(:typep),              do: :deftypep
+  defp typespec(:opaque),             do: :defopaque
+  defp typespec(:spec),               do: :defspec
+  defp typespec(:callback),           do: :defcallback
+  defp typespec(:macrocallback),      do: :defmacrocallback
+  defp typespec(:optional_callbacks), do: :defoptional_callbacks
+  defp typespec(_),                   do: false
```

Comment:
Also look at the lib/elixir/lib/module.ex file and search for `macrocallback` in there. You will need to do some changes.